index.mjs

```diff
@@ -69,14 +78,23 @@ export function $(pieces, ...args) {
 }
 
 $.verbose = true
-// Try `command`, should cover all Bourne-like shells.
-// Try `which`, should cover most other cases.
-// Try `type` command, if the rest fails.
-$.shell = `${execSync('command -v bash || which bash || type -p bash')}`.trim()
-$.prefix = 'set -euo pipefail;'
-$.quote = shq
 $.cwd = undefined
 
+$.quote = (s) => s
+$.prefix = ''
+if ($.shell = which.sync('bash', { nothrow: true })) {
```

Comment:
Now this looks for `bash` without needing Unix-specific commands. The big FIXME is that you can't be really sure the `bash` you find is what you want on Windows, because:
* WSL installs a `bash` executable, which runs commands at the current directory in Linux.
* MSYS2 and Cygwin both have a `bash` executable, and they will be hit when node is running under them. The issue is their argument passing is broken.

I have a patch for Cygwin to fix that behavior, but I am guilty of not following up with the upstream well enough. Heck, I am not even sure if it works.